### ISSUE TITLE
ESlint incorrectly reports indentation error with inline fat arrow function

### ISSUE THREAD

**Issue by roopakv** [eslint/eslint#9686]
* **ESLint Version: 4.4**
* **Node Version: 6.12**
* **npm Version: 3.10.10**

**What parser (default, Babel-ESLint, etc.) are you using?**
default
<details>
<summary>Configuration</summary>

```js
'use strict';

module.exports = {
  extends: 'eslint:recommended',
  env: {
    es6: true,
    mocha: true,
    node: true
  },
  plugins: [
    'mocha',
    'jsdoc'
  ],
  root: true,
  rules: {
    'array-callback-return': 'error',
    'arrow-parens': 'error',
    'block-scoped-var': 'error',
    curly: 'error',
    'dot-location': ['error', 'property'],
    eqeqeq: 'error',
    'max-depth': ['warn', 3],
    'mocha/no-exclusive-tests': 'error',
    'mocha/no-return-and-callback': 'error',
    'no-alert': 'error',
    'no-eval': 'error',
    'no-lone-blocks': 'error',
    'no-shadow-restricted-names': 'error',
    'no-sync': 'warn',
    'no-throw-literal': 'error',
    'no-unmodified-loop-condition': 'error',
    'no-useless-concat': 'error',
    'no-useless-escape': 'off',
    'no-var': 'warn',
    'no-with': 'error',
    'object-property-newline': ['error', {
      allowMultiplePropertiesPerLine: true
    }],
    'vars-on-top': 'error',
    yoda: 'error',
    semi: 'error',
    'valid-jsdoc': ['error',
      {
        requireReturn: false,
        requireReturnDescription: false,
        requireParamDescription: false
      }
    ],
    camelcase: 'off',
    'no-multi-spaces': 'error',
    'quote-props': ['error', 'as-needed'],
    'comma-spacing': ['error',
      {
        before: false,
        after: true
      }
    ],
    'key-spacing': ['error',
      {
        beforeColon: false,
        afterColon: true
      }
    ],
    'comma-dangle': ['error', 'never'],
    'comma-style': ['error', 'last'],
    'sort-imports': 'error',
    'spaced-comment': ['error', 'always'],
    'keyword-spacing': ['error',
      {
        overrides: {
          if: {
            before: true
          },
          else: {
            before: true
          },
          catch: {
            before: true
          }
        }
      }
    ],
    strict: ['error', 'global'],
    'operator-linebreak': ['error', 'after'],
    indent: ['error', 2, {
      SwitchCase: 1,
      MemberExpression: 1,
      FunctionDeclaration: {body: 1, parameters: 2},
      FunctionExpression: {body: 1, parameters: 2}
    }],
    'max-len': ['error', 80, {ignoreComments: true}],
    quotes: ['error', 'single'],
    'no-multi-str': 'error',
    'no-mixed-spaces-and-tabs': 'error',
    'no-trailing-spaces': 'error',
    'space-unary-ops': ['error',
      {
        nonwords: false,
        overrides: {}
      }
    ],
    'one-var': ['error', 'never'],
    'brace-style': ['error', '1tbs'],
    'space-infix-ops': 'error',
    'space-before-blocks': ['error', 'always'],
    'eol-last': 'error',
    'space-before-function-paren': [
      'error',
      {
        anonymous: 'ignore',
        named: 'never'
      }
    ],
    'array-bracket-spacing': [
      'error',
      'never'
    ],
    'space-in-parens': ['error', 'never'],
    'no-multiple-empty-lines': 'error',
    'no-spaced-func': 'error',
    'dot-notation': 'error',
    'no-console': 'error',
    'arrow-spacing': ['error', { before: true, after: true }],
    // jsdoc rules
    'jsdoc/check-types': 'error',
    'jsdoc/check-param-names': 'error',
    'jsdoc/check-tag-names': 'error',
    'jsdoc/newline-after-description': 'error',
    'jsdoc/require-param': 'error',
    'jsdoc/require-param-type': 'error',
    'jsdoc/require-returns-type': 'error'
  },
  settings: {
    jsdoc: {
      tagNamePreference: {
        returns: 'return'
      }
    }
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
'use strict';

const something = require('something');
const sinon = require('sinon');

let someInterface;
let somethingElse;

function thisIsAnExample(paramWhichIsSooperLong, someKindOfId,
    anotherKindOfId) {
  return something.someMethod(paramWhichIsSooperLong,
    someKindOfId, anotherKindOfId).then(() => {
      sinon.assert.calledOnce(someInterface.something);
      sinon.assert.calledWith(somethingElse.test);
  });
}

module.exports = thisIsAnExample;
```

```bash
eslint lint_example.js
```

**What did you expect to happen?**
I expected the lint to be successful and not throw errors
**What actually happened? Please include the actual, raw output from ESLint.**
ESlint wants to un-indent the lines starting with sinon. Running it with --fix actually does this.

```
> eslint lint_example.js


/Users/roopak/workspace/spoke2/lint_example.js
  13:1  error  Expected indentation of 4 spaces but found 6  indent
  14:1  error  Expected indentation of 4 spaces but found 6  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

```

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, this is working as intended -- `.then` is a method call on the `something.someMethod(...)` result and not one of the arguments of `something.someMethod`, so it is offset from the `something.someMethod` text.

This is more apparent if you add a linebreak after the `something.someMethod(...)` call:

```js
  return something.someMethod(paramWhichIsSooperLong,
    someKindOfId, anotherKindOfId
  ).then(() => {
      // The following lines should only be offset by 2 spaces.
      sinon.assert.calledOnce(someInterface.something);
      sinon.assert.calledWith(somethingElse.test);
  });
```

**Comment by roopakv** [comment]
Hmm thank you for the quick response. Is there any rule which can auto fix it to make the then on a new line? We're migrating from eslint 3.x and the auto fix here is causing a snafu. Or is the only way to do this manually?

i did look at newline-per-chained-call but we want something where it doesnt use a new line if it all fits in one line.

**Comment by not-an-aardvark** [comment]
Sorry about the delayed response. You could try `function-paren-newline`, although I'm not sure if it would do exactly what you're looking for.

Closing because this is working as intended, but feel free to reply if you have any more questions.

**Comment by roopakv** [comment]
I understand where you are coming from. i did try the `function-paren-newline` but it didnt actually help. We're still on legacy indent, and im trying to see if there is a better way for us to move rather than to suddenly have a PR make changes across hundreds of files.
